### PR TITLE
[Merged by Bors] - Add sip exec workaround for skipped processes

### PR DESCRIPTION
**Pull request by DmitryDodzin**
relevant for #956 with `--skip-processes cargo-watch`

### CODE REVIEW

File: mirrord/layer/src/load.rs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+}
+
+pub fn load_type(given_process: &str, config: LayerConfig) -> LoadType {
+    let skip_processes = config.skip_processes.clone().map(VecOrSingle::to_vec);
+
+    if should_load(given_process, skip_processes) {
+        LoadType::Full(Box::new(config))
+    } else {
+        #[cfg(target_os = "macos")]
+        if sip::is_sip_exec(given_process) {
```
Comment (aviramha) [inline mirrord/layer/src/load.rs:29]: I don't agree with the terminology - I think this should be `is_sip_only`

File: mirrord/layer/src/load.rs [hunk 2]
```diff
@@ -0,0 +1,77 @@
+use mirrord_config::{util::VecOrSingle, LayerConfig};
+
+#[cfg(target_os = "macos")]
+mod sip {
```
Comment (aviramha) [inline mirrord/layer/src/load.rs:4]: Add docs explaining this (why we do that)

File: mirrord/layer/src/load.rs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    use std::{collections::HashSet, sync::LazyLock};
+
+    static SIP_EXEC_PROCESSES: LazyLock<HashSet<&str>> =
+        LazyLock::new(|| HashSet::from(["sh", "bash", "cargo", "rustc", "cargo-watch"]));
```
Comment (aviramha) [inline mirrord/layer/src/load.rs:8]: I don't agree on the sh/bash - it might be beneficial for users to load into those.

**Review by aviramha — CHANGES_REQUESTED** [review]

File: mirrord/layer/src/load.rs (hunk 3 above)
Comment (DmitryDodzin) [reply mirrord/layer/src/load.rs:8]: it's only if specified in `--skip-processes sh`

Comment (aviramha) [reply mirrord/layer/src/load.rs:8]: ah, okay need better docs 🖌️

File: CHANGELOG.md [hunk 4]
```diff
@@ -7,6 +7,11 @@ Check [Keep a Changelog](http://keepachangelog.com/) for recommendations on how
 
 ## [Unreleased]
 
+### Changed
+
+- mirrord-layer: partialy load mirrord on certian processes that spawn other processes to allow sip patch on the spawned process.
```
Comment (aviramha) [inline CHANGELOG.md:12]: add (macOS only)

File: mirrord/layer/src/load.rs [hunk 5]
```diff
@@ -0,0 +1,84 @@
+use mirrord_config::{util::VecOrSingle, LayerConfig};
+
+/// For processes that spawn other processes whe should also patch sip on the spawned instances,
```
Comment (aviramha) [inline mirrord/layer/src/load.rs:3]: we*

File: mirrord/layer/src/load.rs [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+/// curretly limiting to list from `SIP_ONLY_PROCESSES`
+#[cfg(target_os = "macos")]
```
Comment (aviramha) [inline mirrord/layer/src/load.rs:5]: Add a comment explaining that this list is being hit only when it's also in the `skip processes` list

**Review by aviramha — APPROVED** [review]
gj!
bors r+

**Pull request closed without merging**